src/engine/QueryPlanner.cpp

```diff
@@ -1159,17 +1161,21 @@ Variable QueryPlanner::generateUniqueVarName() {
 }
 
 // _____________________________________________________________________________
-QueryPlanner::SubtreePlan QueryPlanner::getTextLeafPlan(
+vector<QueryPlanner::SubtreePlan> QueryPlanner::getTextLeafPlan(
     const QueryPlanner::TripleGraph::Node& node) const {
-  SubtreePlan plan(_qec);
-  plan._idsOfIncludedNodes |= (size_t(1) << node._id);
-  auto& tree = *plan._qet;
+  vector<SubtreePlan> vecPlans;
   AD_CONTRACT_CHECK(node._wordPart.has_value());
-  auto textOp = std::make_shared<TextOperationWithoutFilter>(
-      _qec, node._wordPart.value(), node._variables, node._cvar.value());
-  tree.setOperation(QueryExecutionTree::OperationType::TEXT_WITHOUT_FILTER,
-                    textOp);
-  return plan;
+  for (string word : node._wordPart.value()) {
+    SubtreePlan plan(_qec);
+    plan._idsOfIncludedNodes |= (size_t(1) << node._id);
```

Comment:
This is the problem here:
Each of your different scans (for the different subwords) needs to have its own Id, otherwise only one of them will be executed, and they will never be joined together.

Reply:
You probably have to get additional Ids for the `node` inside of the query graph, have a look at that mechanism and then we can think of something when you are back.